settings.gradle.kts

```diff
@@ -1,12 +1,84 @@
+            }
+
+            /*
+             * This property group is used to build kotlinx.atomicfu against Kotlin compiler snapshots.
+             * When build_snapshot_train is set to true, kotlin_version property is overridden with kotlin_snapshot_version.
+             * Additionally, mavenLocal and Sonatype snapshots are added to repository list
+             * (the former is required for AFU and public, the latter is required for compiler snapshots).
+             * DO NOT change the name of these properties without adapting kotlinx.train build chain.
+             */
+            val buildSnapshotTrainGradleProperty = providers.gradleProperty("build_snapshot_train")
+            if (buildSnapshotTrainGradleProperty.isPresent) {
+                maven(url = uri("https://oss.sonatype.org/content/repositories/snapshots"))
+            }
+
+        }
+    }
+}
 
 dependencyResolutionManagement {
+    val buildSnapshotTrainGradleProperty = providers.gradleProperty("build_snapshot_train")
+
+    @Suppress("UnstableApiUsage")
+    repositoriesMode.set(RepositoriesMode.PREFER_SETTINGS)
+
+    @Suppress("UnstableApiUsage")
+    repositories {
+
+        // TODO(Dmitrii Krasnov): maybe could be passed via additionalRepositoryProperty
+        maven("https://maven.pkg.jetbrains.space/kotlin/p/kotlin/dev")
+
+        if (buildSnapshotTrainGradleProperty.isPresent) {
+            maven(url = uri("https://oss.sonatype.org/content/repositories/snapshots"))
+        }
+
+        val additionalRepositoryProperty = providers.gradleProperty("community.project.kotlin.repo")
+            .orElse(providers.gradleProperty("kotlin_repo_url"))
+        if (additionalRepositoryProperty.isPresent) {
+            maven(url = uri(additionalRepositoryProperty.get()))
+            logger.info("A custom Kotlin repository ${additionalRepositoryProperty.get()} was added")
+        }
+
+        mavenCentral()
+
+        exclusiveContent {
+            forRepository {
+                ivy("https://nodejs.org/download/v8-canary") {
+                    patternLayout { artifact("v[revision]/[artifact](-v[revision]-[classifier]).[ext]") }
+                    metadataSources { artifact() }
+                    content { includeModule("org.nodejs", "node") }
+                }
+            }
+            filter { includeGroup("org.nodejs") }
+        }
+
+        exclusiveContent {
+            forRepository {
+                ivy("https://github.com/yarnpkg/yarn/releases/download") {
+                    patternLayout { artifact("v[revision]/[artifact](-v[revision]).[ext]") }
+                    metadataSources { artifact() }
+                    content { includeModule("com.yarnpkg", "yarn") }
+                }
+            }
+            filter { includeGroup("com.yarnpkg") }
+        }
+
+        // TODO(Dmitrii Krasnov): I'm not sure that we really need it
+        mavenLocal()
```

Comment:
Yeah, it's rare that Maven Local is needed, so it should probably be removed. If a dev needs it, they can add it manually. What do you think? Should it be removed in this PR, or a later one?

Reply:
I see that [it was added recently](https://github.com/Kotlin/kotlinx-atomicfu/pull/392) (Jan 2024) for train builds. I'm trying to figure out is it still really necessary for train builds.
I hope that I figure out during this PR, but I believe that we can postpone removing it until the next PR.

Reply:
The only configuration where we used `mavenLocal` was `train build`. I've checked whether it is still necessary and removed it.